Automattic/jetpack
# Social: Add SIG toggle for Jetpack Settings

This change adds the ability to be able to toggle SIG and select default templates from the Jetpack Settings page. This is needed so people with only the main Jetpack plugin can access this setting too.

<img width="1068" alt="CleanShot 2023-08-15 at 12 45 44 png" src="https://github.com/Automattic/jetpack/assets/36671565/3c2227dd-6f03-434a-b33b-85b138bc436d">


## Proposed changes:

* Restructured the SIG components a bit
* Registering the Social store is now happening in `publicize-components`
* I added 2 new components for opening the template picker with a button, and a generic toggle for SIG. We can use these on the Social admin page too in a next change
* I added these components into _inc under the main Publicize toggle.

### Other information:


## Jetpack product discussion


1205142973434174-as-1203857426851211
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Go to `wp-admin/admin.php?page=jetpack#/sharing` on your site, and compare the looks with the image above
* You should see the SIG toggle and the template picker only if
  * You have the publicize module enabled with the main toggle
  * You have the Advanced plan - the only way currently to have the feature
* Test the functionality
  * Toggling the state you work, and stay the same on refresh
  * Changing templates should stay the same on refresh
  * You can open the Social admin page too, change it there, then refresh the Jetpack settings page to see if the changes went throug.

* A generic sanity testing would be nice too, as the store structure is changed a bit.
  * Create a new post and publicize it.
  * Play with the settings in the editor, you should see no errors.
